### PR TITLE
[IMP] web: support addition/subtraction timedelta on datetime.datetime in pyeval. also update domain in abandoned carts on website dashboard in website_sale module

### PR DESCRIPTION
**Pull request by rop-odoo**
Issue: https://www.odoo.com/web?#id=745391&view_type=form&model=project.issue&action=609&menu_id=4720

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/controllers/backend.py [hunk 1]
```diff
@@ -85,9 +85,11 @@ def fetch_dashboard_data(self, date_from, date_to):
                 ('partner_id', '!=', request.env.ref('base.public_partner').id),
                 ('invoice_status', '=', 'to invoice'),
             ]),
-            order_carts_abandoned_count=request.env['sale.order'].search_count(sale_order_domain + [
+            order_carts_abandoned_count=request.env['sale.order'].search_count([
```
Comment (JKE-be) [inline addons/website_sale/controllers/backend.py]: why remove sale_order_domain ? 

Now you will returl all result and not only the interval between date_from and date_to

File: addons/website_sale/models/sale_order.py [hunk 2]
```diff
@@ -22,6 +22,12 @@ class SaleOrder(models.Model):
     payment_acquirer_id = fields.Many2one('payment.acquirer', string='Payment Acquirer', copy=False)
     payment_tx_id = fields.Many2one('payment.transaction', string='Transaction', copy=False)
     only_services = fields.Boolean(compute='_compute_cart_info', string='Only Services')
+    is_public_user = fields.Boolean(string="Is Public User", compute='_compute_is_public_user', store=True)
```
Comment (JKE-be) [inline addons/website_sale/models/sale_order.py]: why add this field  ? 

not acceptable in stable version, and probably overkill in master 

If it is just for the domain below, you can just use eval.

File: addons/website_sale/views/sale_order_views.xml [hunk 3]
```diff
@@ -49,8 +47,8 @@
         <field name="name">Abandoned Carts</field>
         <field name="res_model">sale.order</field>
         <field name="view_mode">tree,form</field>
-        <field name="domain">[('state', '=', 'draft'), ('order_line', '!=', False), ('date_order', '&lt;=', (context_today() - datetime.timedelta(hours=1)).strftime('%Y-%m-%d'))]</field>
-        <field name="context">{'show_sale': True, 'search_default_from_website': 1, 'create': False}</field>
+        <field name="domain">[('state', '=', 'draft'), ('order_line', '!=', False), ('team_id.team_type', '=', 'website'), ('is_public_user', '=', False), ('date_order', '&lt;=', (datetime.datetime.utcnow() - datetime.timedelta(hours=1)).strftime('%Y-%m-%d %H:%M:%S'))]</field>
```
Comment (JKE-be) [inline addons/website_sale/views/sale_order_views.xml]: 'utcnow' ...  why not 'now' ?

+use SERVER_DATETIME_FORMAT

**Comment by rop-odoo** [comment]
@JKE-be , i done the changes

File: addons/website_sale/models/sale_order.py (hunk 2 above)
Comment (hmo-odoo) [inline addons/website_sale/models/sale_order.py]: @JKE-be , yes you are right, no need to add new field.

and we need to eval domain in client side due to datetime checking
so we use eval in context and pass id of public user partner.

**Comment by hmo-odoo** [comment]
merged at https://github.com/odoo/odoo/commit/477666d3da0d9244c6e974aeee1975eec38a6bb6

**Pull request closed without merging**